lib/scala/project-manager/src/main/scala/org/enso/projectmanager/infrastructure/http/ProjectsEndpoint.scala

```diff
@@ -71,13 +73,14 @@ final class ProjectsEndpoint[
           )
         )
     }
-  }
 
   private def buildEnsoArchive(
-    projectId: UUID
+    projectId: UUID,
+    subdirectory: Option[String]
   ): Future[Either[ProjectRepositoryFailure, Option[EnsoProjectArchive]]] =
     Exec[F].exec {
-      projectRepository
+      projectRepositoryFactory
+        .getProjectRepository(None, subdirectory.map(new File(_)))
```

Comment:
No need for additional parameters. You can pass the `projectsDirectory` directly